### DISCUSSION TITLE
Notification when Pull Requests gets in conflict state

### DISCUSSION THREAD

**Discussion by cbjeukendrup** [community/community#4335] [Pull Requests]
I thought that it would be very useful to get a notification (email) when a Pull Request becomes in a conflict state (unmergeable) due to other pull requests being merged or changes to the base branch. This would allow the author of the pull request to rebase it soon, so that it can be merged. 

Currently, it's easy to miss when a PR becomes unmergeable. As far as I know, the only way to notice it, is manually checking the pull request page, or being messaged by a maintainer asking you to rebase the PR. It's currently very possible to be in the situation that the author thinks "Why is my PR not merged?" and the maintainer thinks "Why doesn't the author solve the conflict?", and the answer to both questions is that the author simply does not know about the conflict. This is of course a suboptimal situation. 

Would a notification about this be technically possible? Of course the "simple" way is: when a commit is made to a branch, go over all pull requests of which that branch is the base branch, check for mergeability and send an email when the check is negative. But that might require a lot of power (especially for big repos with many open pull requests...). I don't know how "expensive" such a mergeability check is.

**Comment by tcharding** [comment]
I'd love this.

**Comment by cbjeukendrup** [comment]
If it is difficult/impossible to check 'mergeability' for every open PR after every push, then the following compromis could help: if anyone opens the PR page, the mergeability needs to be checked anyway, to show it at the bottom of the page. It would be great if the author is notified at least when the mergeability check happens anyway.

**Comment by yuki2006** [comment]
I want this feature now.

Curiously, Slack's GitHub app gives me conflict notifications.

**Comment by 0R-gb** [comment]
Is the above question too complicated or unsolvable and therefore ignored? This is a real life problem @github

**Comment by marc-hb** [comment]
According to https://github.com/isaacs/github/issues/224, there is already an API that returns the mergeable status and Slack notifications are already available... if you use Slack.

Clearly, there is already some sort of job that updates the mergeable status so why can't that be hooked to some notification? It's one of the most important state changes for a PR.

GH sends half a dozen notifications by default every time someone or something blinks or sneezes, so it's really hard to understand why this can't be done. 

EDIT: (old) analysis explains that conflict status is computed "lazily":
- https://github.com/pullreminders/backlog/issues/42#issuecomment-436412823

**Reply by eranl** [reply]
> Clearly, there is already some sort of job that updates the mergeable status

Another indication of this is that on the PR page, conflict status shows with almost no delay.

**Comment by Al2Klimov** [comment]
I missed that too, so I've written myself a bot: https://github.com/Al2Klimov/mergeconflict-tgbot

You can self-host your own instance or use mine (linked under _About_).

**Comment by eranl** [comment]
Short of an email notification, even a simple indicator on the PR list page, similar to the test status one, would be helpful, in repos where I have multiple open PRs.
